unity-client/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarAnimatorLegacy.cs

```diff
@@ -153,22 +153,28 @@ void State_Init(BlackBoard bb)
 
     void State_Ground(BlackBoard bb)
     {
+        if (bb.deltaTime <= 0)
+        {
+            Debug.LogError("deltaTime should be > 0", gameObject);
+            return;
+        }
+
         animation[baseClipsIds.run].normalizedSpeed = bb.movementSpeed / bb.deltaTime * bb.runSpeedFactor;
         animation[baseClipsIds.walk].normalizedSpeed = bb.movementSpeed / bb.deltaTime * bb.walkSpeedFactor;
 
         float movementSpeed = bb.movementSpeed / bb.deltaTime;
 
         if (movementSpeed > runMinSpeed)
         {
-            animation.Play(baseClipsIds.run);
+            animation.CrossFade(baseClipsIds.run, RUN_TRANSITION_TIME);
```

Comment:
Transition times still feel pretty abrupt. Specially the Walk -> Idle transition.

I see that the transition times are set to 0.01. We should try to aim for 0.1 or 0.15.